blocks/subscriptions-block/components/PaymentForm/index.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from 'react';
+import Sales from '@arc-publishing/sdk-sales';
+import {
+  CardElement, useElements,
+} from '@stripe/react-stripe-js';
+import {
+  PrimaryFont,
+  Button,
+  BUTTON_SIZES,
+  BUTTON_STYLES,
+} from '@wpmedia/shared-styles';
+
+import './styles.scss';
+
+const CARD_ELEMENT_OPTIONS = {
+  style: {
+    base: {
+      color: '#575757',
+      fontSize: '16px',
+    },
+  },
+};
+
+function PaymentForm({
+  orderNumber,
+  successURL,
+  paymentMethodID,
+  clientSecret,
+  stripeInstance,
+  formTitle,
+  formLabel,
+  submitText,
+}) {
+  const [formStatus, setFormStatus] = useState('idle');
+
+  // stripe hooks have to be within Elements wrapper
+  // https://stripe.com/docs/stripe-js/react#useelements-hook
+  const elements = useElements();
+
+  const handleSubmit = async (
+    event,
+  ) => {
+    event.preventDefault();
+
+    setFormStatus('processing');
+    const cardElement = elements.getElement('card');
+
+    const { error, paymentMethod } = await stripeInstance.createPaymentMethod({
+      type: 'card',
+      card: cardElement,
+      billing_details: {}, // todo: collect other info? eg. name, email, etc.
+    });
+
+    if (error) {
+      setFormStatus('error');
+      return;
+    }
+
+    let result;
+
+    // if order of $0 there's a different stripe logic
+    const totalOrder = Sales.currentOrder.total;
+
+    if (totalOrder > 0) {
+      result = await stripeInstance.confirmCardPayment(clientSecret, {
+        payment_method: paymentMethod.id,
+      });
+    } else {
+      result = await stripeInstance.confirmCardSetup(clientSecret, {
+        payment_method: paymentMethod.id,
+      });
+    }
+
+    if (result.error) {
+      setFormStatus('error');
+      return;
+    }
+
+    if (totalOrder > 0) {
+      const nonZeroPriceOutput = await Sales.finalizePayment(
+        orderNumber,
+        paymentMethodID,
+        // using paymentIntent here for greater than 0
+        result.paymentIntent.id,
+      );
+      if (nonZeroPriceOutput.status === 'Paid') {
+        setFormStatus('success');
+        window.location.href = successURL;
+      } else {
+        setFormStatus('error');
```

Comment:
I can add visual error-handling. I think we have some sort of pre-built components for this

Reply:
Updated with visual error display

<img width="774" alt="Screen Shot 2021-12-20 at 13 23 47" src="https://user-images.githubusercontent.com/5950956/146821508-32379e0f-3df6-4014-9e58-61c24ff613d4.png">